alphabase/io/hdf.py

```diff
@@ -498,9 +498,16 @@ def __init__(
             >>> hdf_file.dfs.df1.data_from
             "colleagues"
         """
-        mode = "w" if delete_existing else "a"
-        with h5py.File(file_name, mode):  # , swmr=True):
+        if delete_existing:
+            mode = "w"
+        elif read_only:
+            mode = "r"
+        else:
+            mode = "a"
```

Comment:
The `truncate` param in the function may become ambiguis

Reply:
I have the same feeling.
I think the naming is also very ambigous.

Reply:
not really sure what you mean? is there action required?

Reply:
For this function, (how to ref a code block??)
        Args:
            file_name (str): file path.
            read_only (bool, optional): If hdf is read-only. Defaults to True.
            truncate (bool, optional): If existing groups and datasets can be
                truncated (i.e. are overwitten). Defaults to False.
            delete_existing (bool, optional): If the file already exists,
                delete it completely and create a new one. Defaults to False.